api/v1alpha1/cappconfig_types.go

```diff
@@ -0,0 +1,51 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// CappConfigSpec defines the desired state of CappConfig
+type CappConfigSpec struct {
+	DNSConfig       DNSConfig       `json:"dnsConfig,omitempty"`
+	AutoscaleConfig AutoscaleConfig `json:"autoscalerConfig,omitempty"`
+}
+
+type DNSConfig struct {
+	Name string            `json:"name,omitempty"`
+	Data map[string]string `json:"data,omitempty"`
```

Comment:
We know what the format of the data in the DNSCONFIG is, so it should not be map[string]string
also, this is required so it should not have the `omitempty` struct tag